.evergreen/compile-unix.sh

```diff
@@ -229,7 +229,7 @@ pkg-config --modversion libssl || true
 
 if [ "$COMPILE_LIBMONGOCRYPT" = "ON" ]; then
    # Build libmongocrypt, using the previously fetched installed source.
-   git clone https://github.com/mongodb/libmongocrypt --branch 1.6.0
+   git clone https://github.com/mongodb/libmongocrypt --branch 1.7.0-alpha0
 
```

Comment:
```suggestion
   git clone https://github.com/mongodb/libmongocrypt
   cd libmongocrypt
   # TODO(CDRIVER-4394) update to use libmongocrypt 1.7.0 once a release is available with a fix to MONGOCRYPT-501.
   git checkout c1dbee1112221568c8568d1804624fa5d69a2d4e
   cd ..
```

The debug-compile-asan-openssl-cse task is failing due to a bug in libmongocrypt: MONGOCRYPT-502. MONGOCRYPT-502 was fixed on master, but not part of the 1.7.0-alpha0 release.

The suggested change is to get the fix to libmongocrypt compile. Hopefully, that will enable the `*-cse` test tasks to run for this PR.